src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -652,9 +650,6 @@ private async Task ProcessRequests<TContext>(IHttpApplication<TContext> applicat
 
                 if (HasStartedConsumingRequestBody)
                 {
-                    RequestBodyPipe.Reader.Complete();
-
-                    // Wait for Http1MessageBody.PumpAsync() to call RequestBodyPipe.Writer.Complete().
                     await messageBody.StopAsync();
```

Comment:
StopAsync now calls PipeReader.Complete() as they need to do different things whether there is a connection pipe or request pipe.